merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/MissionModelGenerator.java

```diff
@@ -646,6 +655,90 @@ public JavaFile generateActivityTypes(final MissionModelRecord missionModel) {
         .build();
   }
 
+  public Pair<JavaFile, List<TypeRule>> generateAutoValueMappers(
+      final MissionModelRecord missionModel,
+      final List<TypeMirror> recordTypes) {
+    final var typeRules = new ArrayList<TypeRule>();
+
+    final var typeName = missionModel.getAutoValueMappersName();
+
+    final var builder =
+        TypeSpec
+            .classBuilder(typeName)
+            .addAnnotation(
+                AnnotationSpec
+                    .builder(javax.annotation.processing.Generated.class)
+                    .addMember("value", "$S", MissionModelProcessor.class.getCanonicalName())
+                    .build())
+            .addModifiers(Modifier.PUBLIC, Modifier.FINAL);
+
+    for (final var record : recordTypes) {
+      final var methodName = record.toString().replace(".", "_");
+      final var recordTypeElement = elementUtils.getTypeElement(record.toString());
+      final var recordComponents = new ArrayList<Pair<String, TypeMirror>>();
+      for (final var element : recordTypeElement.getEnclosedElements()) {
+        if (!(element instanceof RecordComponentElement el)) continue;
+        recordComponents.add(Pair.of(element.toString(), el.getAccessor().getReturnType()));
+      }
+
+      final var typeRule = new TypeRule(
+          new TypePattern.ClassPattern(ClassName.get(ValueMapper.class), List.of(TypePattern.from(record))),
+          Set.of(),
+          recordComponents
+              .stream()
+              .map(component -> (TypePattern) new TypePattern.ClassPattern(
+                  ClassName.get(ValueMapper.class),
+                  List.of(new TypePattern.ClassPattern((ClassName) ClassName.get(component.getRight()).box(), List.of()))))
+              .toList(),
+          typeName,
+          methodName);
+      typeRules.add(typeRule);
+
+      final var methodBuilder = MethodSpec
+          .methodBuilder(methodName)
+          .addModifiers(Modifier.PUBLIC)
+          .addModifiers(Modifier.STATIC)
+          .returns(ParameterizedTypeName.get(ClassName.get(ValueMapper.class), TypeName.get(record)))
+          .addParameters(
+              recordComponents
+                  .stream()
+                  .map(recordComponent -> ParameterSpec
+                      .builder(
+                          ParameterizedTypeName.get(
+                              ClassName.get(ValueMapper.class),
+                              ClassName.get(recordComponent.getRight()).box()),
+                          recordComponent.getLeft() + "ValueMapper",
```

Comment:
If we have `record A(int a, int b, int c)`, I believe this will produce a method with parameters `(ValueMapper<Integer>, ValueMapper<Integer>, ValueMapper<Integer>)`. D'you think it would be possible to deduplicate parameters so we only need one of any given type?

Reply:
I've reworked the generator to add one parameter per unique type, and to track which record components need which type of value mapper.